### PR TITLE
modules: openthread: platform: Added setting default tx power

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
There were some requests from users for adding the possibility of setting the default Openthread tx output power using kConfig. It is possible by adding the CONFIG_OPENTHREAD_DEFAULT_TX_POWER kConfig and assigning it to the tx_power variable in the radio.c file in the Openthread module.

Added the possibility to set default openthread power using kConfig.

### CODE REVIEW

File: modules/openthread/Kconfig.thread [hunk 1]
```diff
@@ -167,3 +167,11 @@ config OPENTHREAD_MAC_STAY_AWAKE_BETWEEN_FRAGMENTS
 	bool "Stay awake between packet fragments"
 	help
 	  This optimization is done at the expense of power consumption on SED/SSED devices.
+
+config OPENTHREAD_DEFAULT_TX_POWER
+	int "Openthread default tx power in dBm"
```
Comment (rlubos) [inline modules/openthread/Kconfig.thread:172]:
```suggestion
	int "OpenThread default tx power in dBm"
```

File: modules/openthread/Kconfig.thread [hunk 2]
```diff
@@ -167,3 +167,11 @@ config OPENTHREAD_MAC_STAY_AWAKE_BETWEEN_FRAGMENTS
...
+	range -40 20 if NRF_802154_RADIO_DRIVER
+	default 0
+	help
+	  Set the default TX output power [dBm] in radio driver for openthread purpose.
```
Comment (rlubos) [inline modules/openthread/Kconfig.thread:176]:
```suggestion
	  Set the default TX output power [dBm] in radio driver for OpenThread purpose. 
```

**Review by rlubos — DISMISSED** [review]

**Review by Damian-Nordic — DISMISSED** [review]

**Comment by rlubos** [comment]
@ArekBalysNordic There's a compliance error:
>  INITIALISED_STATIC: do not initialise statics to 0
File:modules/openthread/platform/radio.c
Line:90

**Review by rlubos — DISMISSED** [review]

File: modules/openthread/platform/radio.c [hunk 3]
```diff
@@ -83,7 +83,12 @@ static const struct device *const radio_dev =
 	DEVICE_DT_GET(DT_CHOSEN(zephyr_ieee802154));
 static struct ieee802154_radio_api *radio_api;
 
+/* Get the default tx output power from Kconfig */
+#if defined(CONFIG_OPENTHREAD_DEFAULT_TX_POWER)
+static int8_t tx_power = CONFIG_OPENTHREAD_DEFAULT_TX_POWER;
```
Comment (pdgendt) [inline modules/openthread/platform/radio.c:88]: `CONFIG_OPENTHREAD_DEFAULT_TX_POWER` should always be defined (`default 0`), not sure why we need to preprocessor check.

Comment (pdgendt) [reply modules/openthread/platform/radio.c:88]: Which samples fail to build? I assume this is due to another bug?

Comment (ArekBalysNordic) [reply modules/openthread/platform/radio.c:88]: I lost the history of checks here, so I didn't remember. If there isn't any purpose to not include Kconfig.thread in some samples, you're right, there is another bug. Honestly, I don't know all projects that use Open Thread, so I thought there are some that have their own configuration and don't include that file. I can delete that, wait on the CI result and see what's going on.

**Review by carlescufi — APPROVED** [review]

File: modules/openthread/platform/radio.c (hunk 3 above)
Comment (carlescufi) [reply modules/openthread/platform/radio.c:88]: Agreed that there should not be any need for the `#ifdef` here actually. @ArekBalysNordic can you please remove it and we'll see if something fails?

**Review by carlescufi — CHANGES_REQUESTED** [review]
Let's get rid of that `#ifdef` guard. Kconfig integer options are always defined.

**Comment by pdgendt** [comment]
Update `tests/subsys/openthread/radio_stub.c` with:

```c
#define CONFIG_OPENTHREAD_DEFAULT_TX_POWER 0
```

To fix the CI build.

**Comment by ArekBalysNordic** [comment]
> ```c
> #define CONFIG_OPENTHREAD_DEFAULT_TX_POWER 0
> ```

Thanks for your help!

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**